### PR TITLE
SAK-46758 Sorting is wrong on Sign-up

### PR DESCRIPTION
**Pull request by jesusmmp**
https://sakaiproject.atlassian.net/browse/SAK-46758
UserSortNameComparator should be used instead of String sort.

Similar to: https://github.com/sakaiproject/sakai/pull/10129

Here were resolved another sorting problems on other tools
https://github.com/sakaiproject/sakai/pull/8002/files

### CODE REVIEW

File: signup/tool/src/java/org/sakaiproject/signup/tool/jsf/organizer/OrganizerSignupMBean.java [hunk 1]
```diff
@@ -443,6 +448,15 @@ private SignupEventTrackingInfo moveAttendee() throws Exception {
 
 	}
 
+	private static User getUser(String userId) {
```
Comment (adrianfish) [inline signup/tool/src/java/org/sakaiproject/signup/tool/jsf/organizer/OrganizerSignupMBean.java:451]: Why have you made this static? Performance reasons?

File: signup/tool/src/java/org/sakaiproject/signup/tool/jsf/organizer/OrganizerSignupMBean.java [hunk 2]
```diff
@@ -443,6 +448,15 @@ private SignupEventTrackingInfo moveAttendee() throws Exception {
...
+		try {
+			return UserDirectoryService.getUser(userId);
```
Comment (adrianfish) [inline signup/tool/src/java/org/sakaiproject/signup/tool/jsf/organizer/OrganizerSignupMBean.java:453]: Why are you using a cover here? This bean already uses dependency injection in faces-config. Is it for performance reasons?

File: signup/tool/src/java/org/sakaiproject/signup/tool/jsf/organizer/OrganizerSignupMBean.java [hunk 3]
```diff
@@ -489,6 +503,15 @@ public void loadAllAttendees(SignupMeeting meeting) {
 			return;
 		}
 
+		if (!allSignupUsers.isEmpty()) {
+			Collections.sort(allSignupUsers, new Comparator<SignupUser>() {
+				public int compare(SignupUser a1, SignupUser a2) {
+					UserSortNameComparator userComparator = new UserSortNameComparator();
+					return userComparator.compare(getUser(a1.getInternalUserId()), getUser(a2.getInternalUserId()));
```
Comment (adrianfish) [inline signup/tool/src/java/org/sakaiproject/signup/tool/jsf/organizer/OrganizerSignupMBean.java:510]: You could make getUser non static here and use OrganizerSignupMBean.this.getUser.

**Review by adrianfish — CHANGES_REQUESTED** [review]
Just a few questions, really.

File: signup/tool/src/java/org/sakaiproject/signup/tool/jsf/organizer/OrganizerSignupMBean.java (hunk 3 above)
Comment (ottenhoff) [reply signup/tool/src/java/org/sakaiproject/signup/tool/jsf/organizer/OrganizerSignupMBean.java:510]: Also what happens to SignupUser.compareTo ? That should be fixed and that should be the only place the sorting is done. It doesn't make sense to add sorting logic to just the OrganizerSignupMBean.

**Comment by jesusmmp** [comment]
Closing. @feralvarez95 will do another PR.

**Pull request closed without merging**